### PR TITLE
fix for windows path

### LINKED ISSUE #16: Parsing kernel debug arguments manually
Right now we use `serde_yaml` which is quite overkill. So it is better to parse it manually because the file structure that we are using is very simple.

### PR DESCRIPTION
**Pull request by bayedieng**
This should close #16. Someone would have to test it on Windows first.

### CODE REVIEW

File: src/kernel/src/main.rs [hunk 1]
```diff
@@ -174,8 +174,16 @@ fn mount_pfs<G: AsRef<Path>>(fs: &Fs, game: G) -> bool {
...
-    path.push("pfs_image.dat");
+    #[cfg(unix)]
+    {
+        path.push("uroot");
+        path.push("pfs_image.dat");
+    }
+    #[cfg(windows)]
+    {
+        path.push("\\uroot");
+        path.push("\\pfs_image.dat");
```
Comment (ultimaweapon) [inline src/kernel/src/main.rs:185]: This does not look right. `PathBuf` should already handled this for us.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/main.rs (hunk 1 above)
Comment (bayedieng) [reply src/kernel/src/main.rs:185]: When @Mou-Ikkai used the parser it pushed uroot and pf_image.dat as a unix path for some reason, even though the Windows path parses correctly.

Comment (ultimaweapon) [reply src/kernel/src/main.rs:185]: The problem is it was working correctly before we switched the argument parser. My idea is we should write a test for the argument parser to see if it does what we expected.

**Comment by bayedieng** [comment]
Legitimately have no idea

**Pull request closed without merging**